compiler/test/suites/pattern_matching.re

```diff
@@ -222,6 +222,18 @@ describe("pattern matching", ({test}) => {
     "or_match_3",
     "match ([5]) { [a, _] | [_, a, _] | [a] => true, _ => false }",
   );
+  // Aliases
+  assertSnapshot("alias_match_1", "match (true) { _ as p => p }");
+  assertSnapshot("alias_match_2", "match (true) { a as b => a && b }");
+  assertSnapshot("alias_match_3", "match (true) { true | false as p => p }");
```

Comment:
Can we also have a runnable test for this?

Reply:
Done!

Reply:
Looks great 👍